server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -330,6 +353,13 @@ private RolloverResult rolloverDataStream(
                 (builder, indexMetadata) -> builder.put(dataStream.rolloverFailureStore(indexMetadata.getIndex(), newGeneration))
             );
         } else {
+            if (autoShardingResult != null) {
+                LongCounter metricCounter = autoShardingMetricCounters.get(autoShardingResult.type());
+                if (metricCounter != null) {
+                    metricCounter.increment();
+                }
```

Comment:
nit: we can use `computeIfPresent` here

Reply:
This doesn't apply after removing `autoShardingMetricCounters`, right? :thinking: 
The multiple if statement with null check aren't very pretty. I thought about using something like with Optionals.ofNullable/isPresent, but I haven't seen any Optional usage in the codebase, so decide to keep it as is.